Orange/widgets/unsupervised/owdistancematrix.py

```diff
@@ -0,0 +1,370 @@
+
+        self.tablemodel = DistanceMatrixModel()
+        view = self.tableview = QTableView()
+        view.setEditTriggers(QTableView.NoEditTriggers)
+        view.setItemDelegate(TableBorderItem())
+        view.setModel(self.tablemodel)
+        view.horizontalHeader().hide()
+        view.verticalHeader().hide()
+        view.setShowGrid(False)
+        selmodel = SymmetricSelectionModel(view.model(), view)
+        selmodel.selectionChanged.connect(self.commit)
+        view.setSelectionModel(selmodel)
+        self.mainArea.layout().addWidget(view)
+
+        settings_box = gui.widgetBox(self.mainArea, orientation="horizontal")
+
+        self.annot_combo = gui.comboBox(
+            settings_box, self, "annotation_idx", label="Labels: ",
+            orientation="horizontal",
+            callback=self._invalidate_annotations, contentsLength=12)
+        self.annot_combo.setModel(VariableListModel())
+        self.annot_combo.model()[:] = ["None", "Enumeration"]
+        gui.rubber(settings_box)
+        self.report_button = gui.button(
+            settings_box, None, "&Report", callback=self.show_report)
+        self.report_button.setAutoDefault(0)
+        gui.separator(settings_box, 40)
+        gui.auto_commit(
+            settings_box, self, "auto_commit",
+            "Send Selected Data", "Auto send is on", box=None)
+
+    def sizeHint(self):
+        return QSize(800, 500)
+
+    def set_distances(self, distances):
+        self.closeContext()
+        self.distances = distances
+        self.tablemodel.set_data(self.distances)
+        self.selection = []
+
+        self.items = items = distances is not None and distances.row_items
+        annotations = ["None", "Enumerate"]
+        if items and not distances.axis:
+            annotations.append("Attribute names")
+        elif isinstance(items, list) and \
+                all(isinstance(item, Variable) for item in items):
+            annotations.append("Name")
+        elif isinstance(items, Table):
+            annotations.extend(
+                    itertools.chain(items.domain, items.domain.metas))
+        self.annot_combo.model()[:] = annotations
+        if isinstance(items, Table) and items.domain.class_var:
+            self.annotation_idx = 2 + len(items.domain.attributes)
+        else:
+            self.annotation_idx = 1 + len(annotations) == 3
+
+        if items:
+            self.openContext(distances, annotations)
+            self._update_labels()
+            self.tableview.resizeColumnsToContents()
+
+    def _invalidate_annotations(self):
+        if self.distances is not None:
+            self._update_labels()
+
+    def _update_labels(self):
+        var = column = None
+        if self.annotation_idx == 0:
+            labels = None
+        elif self.annotation_idx == 1:
+            labels = [str(i + 1) for i in range(self.distances.shape[0])]
+        elif self.annot_combo.model()[self.annotation_idx] == "Attribute names":
```

Comment:
This can produce an
`IndexError: list index out of range`
when `self.annotation_idx` is loaded from settings and greater than the size of the model

To reproduce: load iris, connect to distances (rows), connect to dist matrix (label=iris), delete widget, change to column distances and connect again.